app/models/trip.rb

```diff
@@ -0,0 +1,87 @@
+class Trip < ActiveRecord::Base
+  validates :duration, :start_station_id, :end_station_id, :start_date, :end_date, :bike_id, :subscription_type, :zip_code, presence: true
+  belongs_to :trip_start, :class_name => 'Station', :foreign_key => 'start_station_id'
+  belongs_to :trip_end,   :class_name => 'Station', :foreign_key => 'end_station_id'
+  belongs_to :condition,  :foreign_key => 'start_date'
+
+  def self.average_trip_duration
+    average(:duration).round(2)
+  end
+
+  def self.longest_ride
+    maximum(:duration)
+  end
+
+  def self.shortest_ride
+    minimum(:duration)
+  end
+
+  def self.most_starting_rides_station
+    station_id = group(:start_station_id).order('count_id Desc').limit(1).count(:id).to_a.last.first
+    Station.find(station_id)
+  end
+
+  def self.most_ending_rides_station
+    station_id = group(:end_station_id).order('count_id Desc').limit(1).count(:id).to_a.last.first
+    Station.find(station_id)
```

Comment:
same here. whenever you have to find a specific record from the database within a class method, you should double check if there's a better way to execute the method